src/Mvc/test/Mvc.FunctionalTests/RealServerBackedIntegrationTests.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Mvc.FunctionalTests;
+
+public class RealServerBackedIntegrationTests : IClassFixture<KestrelBasedWapFactory>
+{
+    public KestrelBasedWapFactory Factory { get; }
+
+    public RealServerBackedIntegrationTests(KestrelBasedWapFactory factory)
+    {
+        Factory = factory;
+    }
+
+    [Fact]
+    public async Task RetrievesDataFromRealServer()
+    {
+        // Arrange
+        var expectedMediaType = MediaTypeHeaderValue.Parse("application/json; charset=utf-8");
+
+        // Act
+        var client = Factory.CreateClient();
+        var response = await client.GetAsync("/");
+        var responseContent = await response.Content.ReadAsStringAsync();
+
+        // Assert
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        Assert.Equal(expectedMediaType, response.Content.Headers.ContentType);
+
+        Assert.Equal(5000, client.BaseAddress.Port);
```

Comment:
This is probably enough to verify that the `TestServer` isn't used unless we happen decide to change its default base address.

We could make this even more reliable by hitting an endpiont that references `httpContext.Features.Get<IConnectionSocketFeature>().Socket` and returns `((IPEndpoint)socket.LocalEndpoint).Port` as part of the response body and then verifying the body contains `"Port": 5000` as expected.

Reply:
> new HttpClient() { BaseAddress = factoryClient.BaseAddress }

I like this. I will add another test to do this and will keep the existing one as it will help with verifying the CreateClient() functionality with real server still works.